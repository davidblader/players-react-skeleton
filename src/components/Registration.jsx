import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import VerticalCenter from './VerticalCenter';

class Registration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      newUser: {},
      success: null,
      error: {},
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const { newUser } = this.state;
    newUser[e.target.name] = e.target.value;
    this.setState({ newUser });
  }

  handleSubmit(e) {
    e.preventDefault();
    fetch('https://players-api.developer.alchemy.codes/api/user', {
      method: 'POST',
      headers: { 'Content-type': 'application/json' },
      body: JSON.stringify(this.state.newUser),
    }).then(resp => resp.json())
      .then((data) => {
        this.setState({ success: data.success });
        if (data.success) {
          this.props.setSession(data.token, data.user);
        } else {
          this.setState({ error: data.error });
        }
      });
  }

  getUserError() {
    const userErrors = {
      'Resource already exists.': 'An account with this email already exists!',
    };

    return userErrors[this.state.error.message] || this.state.error.message;
  }

  render() {
    let errorMessage;
    if (this.state.success === true) {
      return <Redirect to="/roster" />;
    } else if (this.state.success === false) {
      errorMessage = <p className="error-msg">{this.getUserError()}</p>;
    }
    return (
      <VerticalCenter>
        <div id="registration">
          Create an Account
          {errorMessage}
          <form onSubmit={this.handleSubmit}>
            <div><input type="text" name="first_name" placeholder="First name" onChange={this.handleChange} required /></div>
            <div><input type="text" name="last_name" placeholder="Last name" onChange={this.handleChange} required /></div>
            <div><input type="text" name="email" placeholder="Email" onChange={this.handleChange} required /></div>
            <div><input type="password" name="password" placeholder="Password" onChange={this.handleChange} required /></div>
            <div><input type="password" name="confirm_password" placeholder="Confirm password" onChange={this.handleChange} required /></div>
            <div><input type="submit" /></div>
          </form>
        </div>
      </VerticalCenter>
    );
  }
}

Registration.propTypes = {
  setSession: PropTypes.func.isRequired,
};

export default Registration;
