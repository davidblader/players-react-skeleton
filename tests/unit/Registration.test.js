import React from 'react';
import { shallow } from 'enzyme';
import Registration from '../../src/components/Registration';

test('Registration shallow renders', () => {
  const component = shallow(<Registration />);
  expect(component).toMatchSnapshot();
});
