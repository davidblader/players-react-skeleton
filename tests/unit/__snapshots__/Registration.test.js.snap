// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration shallow renders 1`] = `
<AnimalCrossingContainer
  id="registration"
>
  <AnimalCrossingHeader>
    Create an Account
  </AnimalCrossingHeader>
  <Image
    src="tomnook"
  />
  <form
    onSubmit={[Function]}
  >
    <div>
      <span
        className="hide"
      >
        First Name
      </span>
      <input
        id="firstName"
        name="first_name"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
      />
    </div>
    <div>
      <span
        className="hide"
      >
        Last Name
      </span>
      <input
        id="lastName"
        name="last_name"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
      />
    </div>
    <div>
      <span
        className="hide"
      >
        Email
      </span>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
      />
    </div>
    <div>
      <span
        className="hide"
      >
        Password
      </span>
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <div>
      <span
        className="hide"
      >
        Confirm Password
      </span>
      <input
        id="confirmPassword"
        name="confirm_password"
        onChange={[Function]}
        placeholder="Confirm Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="btn-container"
    >
      <input
        className="animal-crossing-btn"
        id="register"
        type="submit"
        value="Register"
      />
    </div>
  </form>
</AnimalCrossingContainer>
`;
